TWLight/applications/views.py

```diff
@@ -841,19 +886,55 @@ def post(self, request, *args, **kwargs):
         # important work for Applications. This includes handling the closing
         # dates for applications and sending email notifications to editors
         # about their applications.
+        batch_update_failed = []
+        batch_update_success = []
         for app_pk in request.POST.getlist('applications'):
             try:
                 app = Application.objects.get(pk=app_pk)
             except Application.DoesNotExist:
                 logger.exception('Could not find app with posted pk {pk}; '
                     'continuing through remaining apps'.format(pk=app_pk))
                 continue
-
-            app.status = status
-            app.save()
-        #Translators: After a coordinator has changed the status of a number of applications, this message appears.
-        messages.add_message(request, messages.SUCCESS,
-            _('Batch update successful.'))
+            if app.partner.authorization_method == Partner.PROXY:
+                total_accounts_available_for_distribution = None
+                if app.specific_stream is not None:
+                    if app.specific_stream.accounts_available is not None:
+                        total_apps_approved_or_sent = User.objects.filter(
```

Comment:
Seems quite computationally expensive to do this check for every application in the list. Could we calculate the number once before looping, and then track how many we have left with a simple variable, rather than re-filtering every loop?

Reply:
Correct me if I'm wrong, to compute it once before the loop is to assume every single approval in a batch is for the same partner, is it not?

Reply:
we could, however, filter out duplicates in the batch and query once for every unique partner/collection which I guess would marginally decrease overhead.

Reply:
well, that would make the whole thing pointless because we then would have to do a redundant `Application.objects.get(pk=app_pk)`.

Reply:
figured out a [workaround](https://github.com/WikipediaLibrary/TWLight/pull/260/files#diff-dba635dfa3cab82282374d6f91d3b958R878)

Reply:
I did forget that we could be batch-editing across multiple partners :)

Reply:
> Per Nikki's feedback on Slack, it would be preferable to take no actions so that the coordinator can decide which applications to approve.

This is going to make the workaround redundant.